### ISSUE TITLE
Generated import paths strips anything preceded by a dot (file.dto.ts ...)

### ISSUE THREAD

**Issue by ChronicStone** [unjs/unimport#306]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v18.16.0
- Nuxt Version:     3.8.2
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  bun@1.0.7
- Builder:          -
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

Here's a stackblitz minimal reproduction :
https://stackblitz.com/edit/nuxt-starter-jxv1vc?file=.nuxt%2Ftypes%2Fimports.d.ts

### Describe the bug

1. A folder 'services' contains a file 'demo.service.ts
2. services forlder is flagged for auto import on config

Produced entry on imports.d.ts : 
```ts
  const someDemoFunction: typeof import('../../services/demo')['someDemoFunction']
```

The '.service' part of the file name have been stripped away

**Comment by niko-chaffinchicas** [comment]
Seeing this issue as well, specifically stripping the `.store` from files being auto-imported in Nuxt 3.8.2.

**Comment by nathanchase** [comment]
I also name my Pinia stores `x.store.ts` and this import bug causes all of the Pinia stores to be untyped as a result.

**Comment by niko-chaffinchicas** [comment]
@ChronicStone @nathanchase Looks like this is a complication from the `resolvePath` function that Nuxt sets in the options [when it calls `generateTypeDeclarations`](https://github.com/nuxt/nuxt/blob/8917e5f564f5be13f4fb9c7ca06fa8864adae050/packages/nuxt/src/imports/module.ts#L187). It may be best to open an issue on Nuxt itself or move this issue there.

I'm gonna take a stab at making a PR that could address this there. 🤞

**Comment by niko-chaffinchicas** [comment]
Still going to open a PR on the Nuxt repo itself, but it doesn't look like I can 100% do what we want. It would be beneficial if an option could be set when calling `toTypeDeclarationItems` that allows you to disable [the default extension stripping logic](https://github.com/unjs/unimport/blob/main/src/utils.ts#L125) in favor of your own.

The issue Nuxt runs into is that the [addon for vue-templates](https://github.com/unjs/unimport/blob/63f8ad0aec56f7e930284ce062de484ee516a5aa/src/addons/vue-template.ts#L69C44-L69C44) calls `options.resolvePath`, but doesn't strip the file extension from the result like `toTypeDeclaration` is noted to do above.

**Comment by niko-chaffinchicas** [comment]
Alternatively, being able to specify option overrides for addons specifically as opposed addons _always_ inheriting the general options would work, as it would allow Nuxt to alter the `resolvePath` method based on whether or not it knows that `unimport` will strip the file extension.

**Comment by kiki-kanri** [comment]
Sorry, the problem is actually related to the [pr](https://github.com/unjs/unimport/pull/303), but after fixing this problem, the original [problem](https://github.com/nuxt/nuxt/issues/24445) reappeared again, all types imported with addServerImportsDir become any.

You can refer to the production in the issue mentioned above.

**Comment by niko-chaffinchicas** [comment]
@kiki-kanri It looks like your issue may have resurfaced because of this line in Nitro?

https://github.com/unjs/nitro/blob/78b3e77b978b51195f313c793cb1cfb1bf594f27/src/build.ts#L178

The most recent change to unimport would disable unimport's stripping of the file extension, since there's a custom `resolvePath` option being set.

**Comment by antfu** [comment]
The extension should be handled by `resolvePath` if provided. Because the previous behavior of always stripping the extension is an implicit mandatory convention that caused the issue described in this PR and there is no way to workaround it. For nitro, it should be fixed on nitro side.

**Comment by productdevbook** [comment]
yes I had this problem again, it adds it again at the end of mjs coming from `node_modules`. I will take a look at this. v3.7.1

**Comment by productdevbook** [comment]
> @kiki-kanriNitro'daki bu çizgi nedeniyle sorununuz yeniden ortaya çıkmış gibi mi görünüyor?
> 
> https://github.com/unjs/nitro/blob/78b3e77b978b51195f313c793cb1cfb1bf594f27/src/build.ts#L178
> 
> İçe aktarma işleminde yapılan en son değişiklik, özel bir seçenek ayarlandığından, içe aktarma işleminin dosya uzantısını kaldırmasını devre dışı bırakacaktır `resolvePath`.

i am send fix pr https://github.com/unjs/nitro/pull/2139
